packages/blade/src/components/ProgressBar/CircularProgressBar.web.tsx

```diff
@@ -0,0 +1,173 @@
+import getIn from '~utils/lodashButBetter/get';
+import BaseBox from '~components/Box/BaseBox';
+import { castWebType } from '~utils';
+import { makeMotionTime } from '~utils/makeMotionTime';
+import type { TextProps } from '~components/Typography';
+import { Text, getTextProps } from '~components/Typography';
+import { Svg, Circle } from '~components/Icons/_Svg';
+import getBaseTextStyles from '~components/Typography/BaseText/getBaseTextStyles';
+
+const pulseKeyframes = keyframes`
+  0% {
+    opacity: 100;
+  }
+  50% {
+    opacity: 0.65;
+  }
+  100% {
+    opacity: 100;
+  }
+`;
+
+const getPulseAnimationStyles = ({
+  duration,
+  easing,
+  progressPercent,
+  isMeter,
+}: {
+  duration: string;
+  easing: string;
+  progressPercent: number;
+  isMeter: boolean;
+}): FlattenSimpleInterpolation => {
+  if (isMeter || progressPercent === 100) {
+    return css`
+      height: 100%;
+      width: 100%;
+      transition: stroke-dashoffset ${duration} ${easing};
+    `;
+  }
+
+  return css`
+    height: 100%;
+    width: 100%;
+    opacity: ${pulseAnimation.opacityInitial};
+    background-color: ${pulseAnimation.backgroundColor};
+    transition: stroke-dashoffset ${duration} ${easing};
+    animation: ${pulseKeyframes} ${duration} ${easing} infinite;
+  `;
+};
+
+const StyledText = styled.text<Pick<TextProps<{ variant: 'body' }>, 'size' | 'weight'>>(
+  ({ theme, size, weight }) => {
+    return getBaseTextStyles({
+      theme,
+      ...getTextProps({ variant: 'body', size, weight }),
+    });
+  },
+);
+
+const DeterminatePulseAnimation = styled.circle<
+  Pick<
+    CircularProgressBarFilledProps,
+    'pulseMotionDuration' | 'pulseMotionDelay' | 'motionEasing' | 'progressPercent' | 'isMeter'
+  >
+>(({ theme, pulseMotionDuration, motionEasing, progressPercent, isMeter }) => {
+  const duration = castWebType(makeMotionTime(getIn(theme.motion, pulseMotionDuration)));
+  const easing = castWebType(getIn(theme.motion, motionEasing));
+
+  return getPulseAnimationStyles({
+    duration,
+    easing,
+    progressPercent,
+    isMeter,
+  });
+});
+
+const CircularProgressBarFilled = ({
+  progressPercent,
+  fillColor,
+  backgroundColor,
+  size = 'small',
+  label,
+  showPercentage = true,
+  isMeter,
+}: CircularProgressBarFilledProps): React.ReactElement => {
+  const hasLabel = label && label.trim()?.length > 0;
+  const strokeWidth = circularProgressSizeTokens[size].strokeWidth;
+  // Size of the enclosing square
+  const sqSize = circularProgressSizeTokens[size].size;
+
+  // SVG centers the stroke width on the radius, subtract out so circle fits in square
+  const radius = (sqSize - strokeWidth) / 2;
+  // Enclose circle in a circumscribing square
+  const viewBox = `0 0 ${sqSize} ${sqSize}`;
+  // Arc length at 100% coverage is the circle circumference
+  const dashArray = radius * Math.PI * 2;
+  // Scale 100% coverage overlay with the actual percent
+  const dashOffset = dashArray - (dashArray * progressPercent) / 100;
```

Comment:
I think the same math is being used in both RN/Web. Let's move this to a shared util?

Reply:
done